lib/blockstore/badger/badger_test_suite.go

```diff
@@ -0,0 +1,307 @@
+package badgerbs
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/filecoin-project/lotus/lib/blockstore"
+	blocks "github.com/ipfs/go-block-format"
+	"github.com/ipfs/go-cid"
+	u "github.com/ipfs/go-ipfs-util"
+
+	"github.com/stretchr/testify/require"
+)
+
+// TODO: move this to go-ipfs-blockstore.
```

Comment:
This test suite needs to be somewhere where it can be reused by all blockstore implementations.